e2e/pollers.go

```diff
@@ -230,3 +235,44 @@ func waitUntilPodDeleted(ctx context.Context, kube *kubeclient, podName string)
 		return err == nil, err
 	})
 }
+
+type Poller[T any] interface {
+	PollUntilDone(ctx context.Context, options *runtime.PollUntilDoneOptions) (T, error)
+}
+
+func pollVMSSOperation[T any](ctx context.Context, vmssName string, pollerOpts *runtime.PollUntilDoneOptions, vmssOperation func() (Poller[T], error)) (*T, error) {
+	var vmssResp T
+	var requestError azure.RequestError
+
+	poller, err := vmssOperation()
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			return nil, fmt.Errorf("unable to complete VMSS operation in allotted time of %s: %w", createVMSSPollingTimeout.String(), err)
+		}
+		return nil, err
+	}
```

Comment:
I took 247-253 outside of the PollImmediateWithContext, since there is nothing that needs to retry here. I am also a supporter of only having the necessary bits inside "loops" for future readability. But if you think there would be a case where _poller, err := vmssOperation()_
needs a retry, I can put it back under PollImmediateWithContext

Reply:
This is actually not correct. The poller could fail but then the operation needs to be re-called in order for retry to work. This has been reverted